opencdc/fromproto_test.go

```diff
@@ -0,0 +1,107 @@
+// Copyright © 2023 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package opencdc
+
+import (
+	"testing"
+
+	opencdcv1 "github.com/conduitio/conduit-commons/proto/opencdc/v1"
+	"github.com/matryer/is"
+	"google.golang.org/protobuf/types/known/structpb"
+)
+
+func TestRecord_FromProto(t *testing.T) {
+	is := is.New(t)
+
+	r1 := &opencdcv1.Record{
+		Position:  []byte("standing"),
+		Operation: opencdcv1.Operation_OPERATION_UPDATE,
+		Metadata:  map[string]string{"foo": "bar"},
+		Key:       &opencdcv1.Data{Data: &opencdcv1.Data_RawData{RawData: []byte("padlock-key")}},
+		Payload: &opencdcv1.Change{
+			Before: &opencdcv1.Data{Data: &opencdcv1.Data_RawData{RawData: []byte("yellow")}},
```

Comment:
Nitpick: for the data fields, we can maybe build combinations (e.g. in a loop), to increase test case coverage.